model/task/db.go

```diff
@@ -141,21 +141,21 @@ var (
 	}
 
 	// Checks if task dependencies are attainable/ have met all their dependencies and are not blocked
-	isAttainable = bson.M{
-		"$ne": []interface{}{bson.M{"$reduce": bson.M{
+	isUnattainable = bson.M{
+		"$reduce": bson.M{
 			"input":        "$" + DependsOnKey,
 			"initialValue": false,
-			"in":           bson.M{"$or": []interface{}{"$$" + bsonutil.GetDottedKeyName("value", DependencyUnattainableKey), "$$" + bsonutil.GetDottedKeyName("this", DependencyUnattainableKey)}},
+			"in":           bson.M{"$or": []interface{}{"$$value", "$$" + bsonutil.GetDottedKeyName("this", DependencyUnattainableKey)}},
```

Comment:
nit: I think `bsonutil.GetDottedKeyName("$$this", DependencyUnattainableKey)` is slightly clearer, since the query language treats `$$this` as a unit